### DISCUSSION TITLE
Getting error on calling token nft endpoint

### DISCUSSION THREAD

**Discussion by amanzeniq** [blockscout/blockscout#9740] [Support]
Hi, I'm running blockscout explorer (V-6) locally and currently it's syncing. I'm trying to call this endpoint "http://0.0.0.0/api?module=account&action=tokennfttx&address=0x9A079c7a61e3A0115b12622313AEE4C1CC10b49d&page=1&offset=10&sort=asc&contractaddress=0x2FB72BB6427A49d06bf15f4c2ed4DD8aCf97F44B" but always getting "500 Internal Server Error" "Something went wrong"
![image](https://github.com/blockscout/blockscout/assets/127725342/7c3fd3ef-de13-4ec8-bc1e-edc245ddb76f)

**Comment by GimluCom** [comment]
Could you provide some more details? Do the other endpoints work? What happens if you go to http://127.0.0.1/api?module=block&action=eth_block_number in your browser?

**Reply by amanzeniq** [reply]
all other endpoints are working fine except this one and on browser it's throwing same error "Something went wrong."

**Reply by amanzeniq** [reply]
Hi Gim, I'm still waiting for your response. Can you please help me in this issue?

**Reply by sl1depengwyn** [reply]
Could you please check the logs and send the error from there?

**Reply by amanzeniq** [reply]
{"time":"2024-05-29T11:58:12.821Z","severity":"error","message":"Error while calling RPC action\"** (Plug.Conn.WrapperError) ** (Ecto.QueryError) unknown bind name `:block` in query:\\n\\nfrom t0 in Explorer.Chain.TokenTransfer,\\n  join: t1 in assoc(t0, :token),\\n  as: :token,\\n  where: t0.block_consensus == true,\\n  where: t1.type == \\\"ERC-721\\\",\\n  where: t0.token_contract_address_hash ==\\n  ^%Explorer.Chain.Hash{byte_count: 20, bytes: <<196, 223, 111, 16, 2, 118, 69, 76, 119, 114, 35, 165, 119, 162, 240, 106, 3, 192, 119, 24>>},\\n  order_by: [asc: t0.block_number, asc: t0.log_index],\\n  limit: ^10,\\n  offset: ^0,\\n  preload: [token: t1]\\n\\n    (ecto 3.11.2) lib/ecto/query/builder.ex:925: Ecto.Query.Builder.count_alias!/2\\n    (explorer 6.6.0) lib/explorer/etherscan.ex:334: Explorer.Etherscan.base_nft_transfers_query/2\\n    (explorer 6.6.0) lib/explorer/etherscan.ex:303: Explorer.Etherscan.list_nft_transfers/3\\n    (block_scout_web 6.6.0) lib/block_scout_web/controllers/api/rpc/address_controller.ex:549: BlockScoutWeb.API.RPC.AddressController.list_nft_transfers/3\\n    (block_scout_web 6.6.0) lib/block_scout_web/controllers/api/rpc/address_controller.ex:203: BlockScoutWeb.API.RPC.AddressController.tokennfttx/2\\n    (block_scout_web 6.6.0) lib/block_scout_web/controllers/api/rpc/address_controller.ex:1: BlockScoutWeb.API.RPC.AddressController.action/2\\n    (block_scout_web 6.6.0) lib/block_scout_web/controllers/api/rpc/address_controller.ex:1: BlockScoutWeb.API.RPC.AddressController.phoenix_controller_pipeline/2\\n    (block_scout_web 6.6.0) lib/block_scout_web/controllers/api/rpc/rpc_translator.ex:126: BlockScoutWeb.API.RPC.RPCTranslator.call_controller/3\\n\"","metadata":{"request_id":"F9PzwOiCbIvm_OAABeOB"}}
**{"time":"2024-05-29T11:58:12.821Z","severity":"info","message":"Sent 500 in 3ms on GET /api?module=account&action=tokennfttx&address=0x9A079c7a61e3A0115b12622313AEE4C1CC10b49d&page=1&offset=10&sort=asc&contractaddress=0xc4dF6F100276454C777223a577a2F06a03C07718","metadata":{"duration":3282,"endpoint":"/api?module=account&action=tokennfttx&address=0x9A079c7a61e3A0115b12622313AEE4C1CC10b49d&page=1&offset=10&sort=asc&contractaddress=0xc4dF6F100276454C777223a577a2F06a03C07718","method":"GET","request_id":"F9PzwOiCbIvm_OAABeOB","status":"500","unit":"microsecond"}}**
{"time":"2024-05-29T11:58:12.859Z","severity":"error","message":"failed to fetch internal transactions for blocks [16302763, 16302764, 16302765, 16302766, 16302767, 16302769, 16302770, 16302771, 16302773, 16302774]: ** (ErlangError) Erlang error: [%{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0x0e186ff8cc5883bfc9305bba2a4682397f033e575763da5c50bf82026cdc1198\", transaction_index: 14}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0x0ffd018a64568703b8676d80fa5ed001d2e42c108e70f819ad76da3548635df2\", transaction_index: 13}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0x5014be7b1b237868c11622a3ea3827adfeb36a1ddbf2b42c5ecc2771c735e1d7\", transaction_index: 11}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0x50822035ceea10c390fa777b2b2b44c8cb598ce6aff409a2fcd5b26091466eb5\", transaction_index: 4}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0x5845d99669f0195f58ac3b2ea78f6159238ca471c81871301fca9aed1bb51b9a\", transaction_index: 6}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0x5d4783136f6a92e2a7bb20b0bf4178cf83ba3f97f1003770665956d0981a2f2e\", transaction_index: 15}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0x6abeeff0d42dc9aa7bdaf09bb2307853e31be404a1aa7907651faf1083ea603b\", transaction_index: 9}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0x75388ec3f208093489334f9a2331f82ff61e8872012b7e50e1b2b94d0c722766\", transaction_index: 10}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0x7ca61e574e3615b0a9213339eaf6c7155cda0434641427b0db77126fceb1a707\", transaction_index: 3}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0x836f5aa1ec88188efb6b40438ce419c069c702b74bcb478b15bd9022b88170e4\", transaction_index: 8}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0x8583a79dd992f1d9719ce4c776d7b8870ded701786765e6d57db4f637008a256\", transaction_index: 2}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0x9475adcc7a4c4f39c3b1f520e8f889e558c5956fd407e395e9389096abe9d1e6\", transaction_index: 1}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0xa377492b28ae6ae59b1a190ecbb6aea62eac1cee0eef6369bcc2a94c46bb5936\", transaction_index: 16}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0xb567ec8f370328ae8beaf45e9d1a17bf472c50e6049d467d19c1bc7701aef98f\", transaction_index: 17}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0xb58e8b2ba028efa6daef57dd1466ab87c6c896ee6f34d786dd7eee6cec048a0b\", transaction_index: 5}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0xdc99d6ab786e43c805599f0e8475ebaf701cbc8c5ecd979e721d8f3f43cf8958\", transaction_index: 7}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0xdfcc529caf3480419d1d2c153444639fd72ee59d13375550617704b129495edc\", transaction_index: 0}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302763, transaction_hash: \"0xf1a137d8ee8405fd6c81327ea40016ff53ed4fac19bd3776daf749b9d0bf1be5\", transaction_index: 12}, message: \"the method debug_traceTransaction does not exist/is not available\"}]","metadata":{"count":10,"error_count":10,"fetcher":"internal_transaction"}}
{"time":"2024-05-29T11:58:12.889Z","severity":"error","message":"failed to fetch internal transactions for blocks [16302751, 16302753, 16302754, 16302755, 16302756, 16302757, 16302758, 16302759, 16302760, 16302762]: ** (ErlangError) Erlang error: [%{code: -32601, data: %{block_number: 16302751, transaction_hash: \"0x277067859a5fb351951ff090987753290a811017e7849aa5b2d402b8377266be\", transaction_index: 0}, message: \"the method debug_traceTransaction does not exist/is not available\"}]","metadata":{"count":10,"error_count":10,"fetcher":"internal_transaction"}}
{"time":"2024-05-29T11:58:12.940Z","severity":"error","message":"failed to fetch internal transactions for blocks [16302741, 16302742, 16302743, 16302744, 16302745, 16302746, 16302747, 16302748, 16302749, 16302750]: ** (ErlangError) Erlang error: [%{code: -32601, data: %{block_number: 16302741, transaction_hash: \"0x0b0bc76fbadd15297609f4bd7e1d5e66cff97fded9bb273f01792edb7f302c44\", transaction_index: 0}, message: \"the method debug_traceTransaction does not exist/is not available\"}, %{code: -32601, data: %{block_number: 16302741, transaction_hash: \"0xd06e1da07655e835da577b8da5898a6b71bf4ba823522863049c0978e9c6f3e8\", transaction_index: 1}, message: \"the method debug_traceTransaction does not exist/is not available\"}]","metadata":{"count":10,"error_count":10,"fetcher":"internal_transaction"}}

**Reply by GimluCom** [reply]
> the method `debug_traceTransaction` does not exist/is not available

That is an error returned by your RPC node. You need to enable `debug` methods.
To enable them on geth: `--http.api=eth,net,web3,debug`

**Reply by amanzeniq** [reply]
I was talking about this error which is bold

**{"time":"2024-05-29T11:58:12.821Z","severity":"info","message":"Sent 500 in 3ms on GET /api?module=account&action=tokennfttx&address=0x9A079c7a61e3A0115b12622313AEE4C1CC10b49d&page=1&offset=10&sort=asc&contractaddress=0xc4dF6F100276454C777223a577a2F06a03C07718","metadata":{"duration":3282,"endpoint":"/api?module=account&action=tokennfttx&address=0x9A079c7a61e3A0115b12622313AEE4C1CC10b49d&page=1&offset=10&sort=asc&contractaddress=0xc4dF6F100276454C777223a577a2F06a03C07718","method":"GET","request_id":"F9PzwOiCbIvm_OAABeOB","status":"500","unit":"microsecond"}}**
